gammapy/utils/nddata.py

```diff
@@ -0,0 +1,512 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst 
+
+import itertools
+from collections import OrderedDict
+import numpy as np
+from astropy.units import Quantity, Unit
+from astropy.table import Table, Column
+
+__all__ = [
+    'NDDataArray',
+    'DataAxis',
+    'BinnedDataAxis',
+]
+
+
+class NDDataArray(object):
+    """ND Data Array
+
+    This class represents a n-dimensional data array. The data is stored as a 
+    `~numpy array`. The data axes are separate classes,
+    `~gammapy.utils.nddata.DataAxis` or 
+    `~gammapy.utils.nddata.BinnedDataAxis`.
+    After this class has been initialized any number of axes and a data array
+    can be added. The axis order follows numpy convention for arrays, i.e. the
+    first index of the data array is the first axis. The axes and data can also
+    passed on initialization. The NDDataArray can be interpolated using several
+    interpolation methods. For example see nddata_demo.ipynb in
+    ``gammapy-extra/notebooks``.
+
+    Parameters
+    ----------
+    axes : OrderedDict, optional
+        Key: axis name, Value: array-like, `~gammapy.utils.nddata.DataAxis`
+    data : array-like, optional
+        Data
+    kwargs : dict
+        Use be subclasses to provide convenient constructors
+    """
+    def __init__(self, axes=None, data=None, **kwargs):
+        self._axes = list()
+        self._data = None
+        self._lininterp = None
+
+        if axes is not None:
+            for key, value in axes.items():
+	            if not isinstance(value, DataAxis):
+	                value = DataAxis(value)
+	            value.name = key
+	            self.add_axis(value)
+
+    	if data is not None:
```

Comment:
There's a problem [here](https://travis-ci.org/gammapy/gammapy/jobs/131370137#L946) and a solution [here](https://www.jetbrains.com/pycharm). 😄